app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt

```diff
@@ -88,8 +90,20 @@ class SearchLocationFragment : Fragment() {
     }
 
     private fun redirectToMain() {
-        val fragmentId = if (fromSearchFragment) R.id.searchFragment else R.id.eventsFragment
-        Navigation.findNavController(rootView).popBackStack(fragmentId, false)
+        if (fromSearchFragment) {
+            Navigation.findNavController(rootView).popBackStack(R.id.searchFragment, false)
+        } else {
+            val bundle = Bundle()
+            bundle.putBoolean(NEW_SEARCH, true)
+            val navOptions = NavOptions.Builder()
+                .setEnterAnim(R.anim.slide_in_left)
+                .setExitAnim(R.anim.slide_out_right)
+                .setPopEnterAnim(R.anim.slide_in_left)
+                .setPopExitAnim(R.anim.slide_out_right)
+                .setPopUpTo(R.id.eventsFragment, true)
+                .build()
+            Navigation.findNavController(rootView).navigate(R.id.eventsFragment, bundle, navOptions)
```

Comment:
Not a good solution. It was better earlier. Can you please find another way for it? Reset looks good.

Reply:
@liveHarshit I have updated a better solution. haha, I was stupid before, this is actually a much simpler issue.